tests/common/util.rs

```diff
@@ -1191,12 +1243,525 @@ impl UCommand {
+    /// for side effects.
+    pub fn stderr(&mut self) -> String {
+        String::from_utf8(self.stderr_bytes()).unwrap()
+    }
+
+    pub fn stderr_all(&mut self) -> String {
+        String::from_utf8(self.stderr_all_bytes()).unwrap()
+    }
+
+    /// Reads, consumes and returns all bytes from child's stderr. If stderr is redirected to stdout
+    /// with [`UCommand::stderr_to_stdout`] then always zero bytes are returned. See also
+    /// [`UChild::stdout_bytes`] for side effects.
+    pub fn stderr_bytes(&mut self) -> Vec<u8> {
+        match self.captured_stderr.as_mut() {
+            Some(output) => output.output_as_bytes(),
+            None if self.raw.stderr.is_some() => {
+                let mut buffer: Vec<u8> = vec![];
+                let stderr = self.raw.stderr.as_mut().unwrap();
+                stderr.read_to_end(&mut buffer).unwrap();
+                buffer
+            }
+            None => vec![],
+        }
+    }
+
+    /// Returns all output from start of the child processes' stderr until now, if we captured the
+    /// output or else returns the output of the child process that can currently be read. See also
+    /// [`UChild::stdout_all_bytes`] for additional information.
+    pub fn stderr_all_bytes(&mut self) -> Vec<u8> {
+        match self.captured_stderr.as_mut() {
+            Some(output) => output.output_all_bytes(),
+            None => self.stderr_bytes(),
+        }
+    }
+
+    /// Reads, consumes and returns the exact amount of bytes from stderr. If stderr is redirect to
+    /// stdout with [`UCommand::stderr_to_stdout`] then always zero bytes are returned.
+    ///
+    /// # Important
+    /// This method blocks indefinitely if the `size` amount of bytes cannot be read.
+    pub fn stderr_exact_bytes(&mut self, size: usize) -> Vec<u8> {
+        match self.captured_stderr.as_mut() {
+            Some(output) => output.output_exact_bytes(size),
+            None if self.raw.stderr.is_some() => {
+                let stderr = self.raw.stderr.as_mut().unwrap();
+                let mut buffer = vec![0; size];
+                stderr.read_exact(&mut buffer).unwrap();
+                buffer
+            }
+            None => vec![],
+        }
+    }
+
+    /// Returns true if the child process is still alive and false otherwise.
+    ///
+    /// # Panics
+    /// If an error occurred during [`Child::try_wait`]
+    pub fn try_alive(&mut self) -> bool {
+        // TODO: return false on error of try_wait ?
+        self.raw.try_wait().unwrap().is_none()
+    }
+
+    /// Returns a [`UChildAssertion`] on which helper functions with assertions can be called.
+    pub fn make_assertion(&mut self) -> UChildAssertion {
+        UChildAssertion::new(self)
+    }
+
+    pub fn make_assertion_with_delay(&mut self, millis: u64) -> UChildAssertion {
+        sleep(Duration::from_millis(millis));
+        self.make_assertion()
+    }
```

Comment:
convenience and few key strokes less. I simplified the method a bit to the line you mentioned.

Reply:
Not sure about saving keystrokes :)
```
child.delay(millis).make_assertion()
child.make_assertion_with_delay(millis)
```

More seriously though, I think we should keep simplifying this as much as possible, so I think this can be removed.

Reply:
Autocompletion is your friend here ;) This is a very common usage pattern, so it makes perfect sense having the separate functions both together in a convenience function. I mean you can do as you like, although I prefer the convenience function.